docs/guides/deployment/overriding-job-variables.md

```diff
@@ -0,0 +1,174 @@
+
+Let's look at how to do that.
+
+## Setup
+Say we have the following repo structure:
+```
+» tree
+.
+├── README.md
+├── requirements.txt
+├── demo_project
+│   ├── daily_flow.py
+```
+
+... and we have some `demo_flow.py` file like this:
+
+```python
+import os
+from prefect import flow, task
+
+@task
+def do_something_important(not_so_secret_value: str) -> None:
+    print(f"Doing something important with {not_so_secret_value}!")
+
+@flow(log_prints=True)
+def some_work():
+    environment = os.environ.get("EXECUTION_ENVIRONMENT", "local")
+    
+    print(f"Coming to you live from {environment}!")
+    
+    not_so_secret_value = os.environ.get("MY_NOT_SO_SECRET_CONFIG")
+    
+    if not_so_secret_value is None:
+        raise ValueError("You forgot to set MY_NOT_SO_SECRET_CONFIG!")
+
+    do_something_important(not_so_secret_value)
+```
+## How to override job variables
+### Using a `prefect.yaml` file
+In this case, let's also say we have the following deployment definition in a `prefect.yaml` file at the root of our repository:
+```yaml
+deployments:
+- name: demo-deployment
+  entrypoint: demo_project/demo_flow.py:some_work
+  work_pool:
+    name: local
+  schedule: null
+```
+
+!!! note
+    While not the focus of this guide, note that this deployment definition uses a default "global" `pull` step, because one is not explicitly defined. For reference, here's what that would look like at the top of the `prefect.yaml` file:
+    ```yaml
+    pull:
+    - prefect.deployments.steps.git_clone: &clone_repo
+        repository: https://github.com/some-user/prefect-monorepo
+        branch: main
+    ```
+
+To provide the `EXECUTION_ENVIRONMENT` and `MY_NOT_SO_SECRET_CONFIG` environment variables to this deployment, we can add a `job_variables` section to our deployment definition in the `prefect.yaml` file:
+
+#### Hard-coded job variables
+```yaml
+deployments:
+- name: demo-deployment
+  entrypoint: demo_project/demo_flow.py:some_work
+  work_pool:
+    name: local
+    job_variables:
+        env:
+        EXECUTION_ENVIRONMENT: staging
+        MY_NOT_SO_SECRET_CONFIG: plumbus
+  schedule: null
+```
+
+... and then run `prefect deploy -n demo-deployment` to deploy the flow with these job variables.
+
+We should then be able to see the job variables in the `Configuration` tab of the deployment in the UI:
+
+![Job variables in the UI](/img/guides/job-variables.png)
+
+#### Using existing environment variables
+If you want to use environment variables that are already set in your local environment, you can template these in the `prefect.yaml` file using the `{{ env.VARIABLE_NAME }}` syntax:
+
+```yaml
+deployments:
+- name: demo-deployment
+  entrypoint: demo_project/demo_flow.py:some_work
+  work_pool:
+    name: local
+    job_variables:
+        env:
+        EXECUTION_ENVIRONMENT: "{{ $EXECUTION_ENVIRONMENT }}"
```

Comment:
same here

Reply:
updated in [72147fb](https://github.com/PrefectHQ/prefect/pull/12033/commits/72147fb9aa10088c247bbca378fa0ba1059be314)